AdminUi/src/AdminUi/ClientApp/src/app/components/quotas/identity/identity-details/identity-details.component.ts

```diff
@@ -217,10 +221,7 @@ export class IdentityDetailsComponent {
             error: (err: any) => {
                 this.loading = false;
                 const errorMessage = err.error?.error?.message ?? err.message;
-                this.snackBar.open(errorMessage, "Dismiss", {
-                    verticalPosition: "top",
-                    horizontalPosition: "center"
-                });
+                this.openAssignQuotaDialog(errorMessage, quotaData);
```

Comment:
I don't get it. Why do you reopen the dialog instead of just not closing it at the first place?

Reply:
I couldn't figure out a better solution than this.. now i implemented another one with service as a transmitter of data between two components.

Reply:
What if you pass a callback function via the `data` of the dialog? I don't like that it lacks type safety, but it's still way better than what's currently implemented.

Reply:
I need something cleared.. do u want to use callback function when the data is sent from dialog to the parent or vise versa or in both cases?

Reply:
If there is an error message, it can be displayed by the dialog. If there is no error message, the dialog can be closed.

Reply:
I added a callback function.